flang/lib/Lower/Bridge.cpp

```diff
@@ -2916,31 +2916,37 @@ class FirConverter : public Fortran::lower::AbstractConverter {
     // with sequences of i1. That is, an array of i1 will be truncated in size
     // and be too small. For example, a buffer of type fir.array<7xi1> will have
     // 0 size.
-    mlir::Type i64Ty = builder->getIntegerType(64);
+    mlir::Type i1Ty = builder->getI1Type();
+    mlir::Type i8Ty = builder->getI8Type();
     mlir::TupleType ty = fir::factory::getRaggedArrayHeaderType(*builder);
-    mlir::Type buffTy = ty.getType(1);
-    mlir::Type shTy = ty.getType(2);
+    mlir::Type buffTy = ty.getType(kBufferPosInHeader);
+    mlir::Type shTy = ty.getType(kExtentPosInHeader);
```

Comment:
This type will be laid out to the _target_ machine (more precisely the target datalayout) and not the build machine's C++ compiler. There is no reason to make assumptions about the relation of those two.

Reply:
Not sure I follow here. I just replaced the hard coded value with a constant so it's easier to change in the future if we need to.

Reply:
I mean the assumption that the build machine's C++ compiler will layout

```c++
struct {
  bool ;
  uint8_t ;
  int64_t * ;
  int64_t * ;
};
```

in precisely the same way in memory as LLVM will layout

```llvm
{ i1, i8, i64*, i64* }
```

and regardless of the target's datalayout is simply not guaranteed.